docs/operate/architecture.mdx

```diff
@@ -0,0 +1,38 @@
+
+## Nodes and roles
+
+You'll see many mentions of server and node throughout this documentation. Generally, we use the term "server" to refer to a running instance of the `restate-server` binary, although in some instances this might also refer to the entire Restate cluster which acts as a single entity. This binary can host multiple functions. When you start a single-node Restate server, for example when doing some local development or testing, you are hosting all the essential features in a single process. These include accepting incoming requests, durably recording events, processing work (delegating invocations to services, handling key-value operations), as well as maintaining metadata used internally by the system.
+
+At its simplest, running a cluster is not that different - multiple nodes cooperate to share the responsibilities we mentioned earlier. This is accomplished by having multiple copies of the server process running on separate machines, although it is possible to create test clusters on a single machine. **Nodes** are therefore distinct instances of the Restate server within a cluster.
+
+Restate clusters are designed to scale out in support of large deployments. As you add more machines, it becomes wasteful to replicate all the functionality across all the machines in a cluster, since not all features need to scale out at the same rate. **Roles** control which features run on any given node, enabling specialization within the cluster.
+
+## Metadata service: the source of truth for cluster-wide information
+
+The Restate **metadata store** is the internal source of truth for node membership and responsibilities. It is essential to the correctness of the overall system: in a cluster this service enables distributed consensus about other components' configuration. All nodes in a Restate cluster must be able to access the metadata service, though not all members of the cluster need to be part of hosting it. Restate includes a built-in Raft-based metadata store which is hosted on all nodes running the `metadata-server` role.
+
+The metadata store is designed to support relatively low volumes of read and write operations (at least compared to other parts of Restate), with the highest level of integrity and availability.
+
+## Ingress: accepting external requests
+
+External requests enter the Restate cluster via the **HTTP ingress** component, which runs on nodes assigned the `http-ingress` role. Compared to other roles, the HTTP ingress role does not involve long-lived state and it can move around relatively freely, since it only handles ongoing client connections.
```

Comment:
For backwards compatibility reasons we don't use the dedicated `http-ingress` role yet. Instead every `worker` role runs the `http-ingress` role in the next release.

Reply:
Let me see if I can weave that in.